### PR TITLE
Release job check for changes fix

### PR DESCRIPTION
**Pull request by squadgazzz**
Check for changes job was wrongly implemented. The PR fixes that.

### CODE REVIEW

**Review by Mateo-mro — APPROVED** [review]

File: .github/workflows/release.yml
Comment (MartinquaXD) [inline .github/workflows/release.yml:32]: Isn't using `diff` technically more correct? Will probably never happen but you could get to the same final state in multiple ways (sequences of commits).
What exactly was the issue with the `diff` command that it led to the wrong result here?

Comment (squadgazzz) [reply .github/workflows/release.yml:32]: I just copied that from the `gnosis/solvers` job. It was used since the release script doesn't really care about the code changes but the commits made since the latest release(no matter the reason they were made).

Comment (squadgazzz) [reply .github/workflows/release.yml]: Switched to `git diff`.

File: .github/workflows/release.yml
Comment (MartinquaXD) [inline .github/workflows/release.yml:5]: This is just for testing the PR and should be removed before merging, right?

**Review by MartinquaXD — APPROVED** [review]
Approved assuming the last discussion gets resolved (either by removing the change or explaining why it's needed)

File: .github/workflows/release.yml
Comment (squadgazzz) [reply .github/workflows/release.yml:5]: Yes

**Pull request merged**
